util/comparator.cc

```diff
@@ -132,7 +132,7 @@ class BytewiseComparatorImpl : public Comparator {
   }
 };
 
-class ReverseBytewiseComparatorImpl : public BytewiseComparatorImpl {
+class ReverseBytewiseComparatorImpl final : public Comparator {
```

Comment:
While making this change I noticed that  `ReverseBytewiseComparatorImpl` used to inherit `BytewiseComparatorImpl::IsSameLengthImmediateSuccessor`. That does not look correct.